### ISSUE TITLE
Deprecation in YAML parser

### ISSUE THREAD

**Issue by prudloff-insite** [symfony/symfony#54704]
### Symfony version(s) affected

6.4.3

### Description

We noticed that the YAML parser triggers a deprecation with some malformed strings:
```
   DEPRECATED  substr(): Passing null to parameter #2 ($offset) of type int is deprecated in /var/www/vendor/symfony/yaml/Parser.php on line 641.
```

### How to reproduce

```php
  $yaml = new \Symfony\Component\Yaml\Parser();
  $yaml->parse(',:
-

');
```

### Additional Context

We noticed this while testing our YAML parsing logic with [php-fuzzer]( https://github.com/nikic/PHP-Fuzzer).

**Comment by GromNaN** [comment]
You you provide a sample Yaml file that produce this error?

**Comment by xabbuh** [comment]
good catch, should be fixed with #54706
